### PR TITLE
Generate inline structs for encoder.

### PR DESCRIPTION
**Pull request by klauspost**
This will generate code for inline structs.

Common Go json -> struct generators  [jsonstruct.com](http://www.jsonstruct.com/) generate code like this:
```Go
type ExampleStruct struct {
	Href     string `json:"href"`
	Kommuner []struct {
		Href string `json:"href"`
		Kode string `json:"kode"`
		Navn string `json:"navn"`
	} `json:"kommuner"`
	Navn      string `json:"navn"`
	Postnumre []struct {
		Href string `json:"href"`
		Navn string `json:"navn"`
		Nr   string `json:"nr"`
	} `json:"postnumre"`
}
```
Here ffjson would fall back to using json.Marshal, but it will now generate inline code for these. All ffjson types are supported, even structs within structs (more inception).

### CODE REVIEW

File: inception/encoder.go [hunk 1]
```diff
@@ -265,6 +266,38 @@ func getGetInnerValue(ic *Inception, name string, typ reflect.Type, ptr bool, fo
 		out += "buf.Write(obj)" + "\n"
 	case reflect.Map:
 		out += getMapValue(ic, ptname, typ, ptr, forceString)
+	case reflect.Struct:
+		if typ.Name() == "" {
+			ic.q.Write("{")
+			ic.q.Write(" ")
+			out += fmt.Sprintf("/* Inline struct. type=%v kind=%v */\n", typ, typ.Kind())
+			newV := reflect.Indirect(reflect.New(typ)).Interface()
+			fields := extractFields(newV)
```
Comment (pquerna) [inline inception/encoder.go]: wow. it goes deeper.

**Comment by pquerna** [comment]
From a quick read I think this looks reasonable, though it needs a rebase.

**Comment by klauspost** [comment]
It *should* be merged correctly with master now.

**Comment by klauspost** [comment]
ok, the diff is really weird. I don't want to risk a bad merge, even if it is a github bug, so I have resubmitted this as a clean PR in #110 (please merge before too long)

**Pull request closed without merging**